x-pack/plugins/apm/public/components/app/Main/FetchStatus.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiDelayHide, EuiPortal, EuiProgress } from '@elastic/eui';
+import React, { Fragment, useMemo, useReducer } from 'react';
+
+export const FetchStatusContext = React.createContext({
```

Comment:
Good point. I also felt that something was off with the naming. This exports two things: a component for rendering the loading indicator (`FetchStatus`), and a context (`FetchStatusContext`).

I can name them something like `GlobalProgress` (component) and `GlobalProgressContext` (context).

Reply:
Or maybe the name of the component and the context doesn't need to match up. So it could be `GlobalFetchIndicator` for the ui component, and `GlobalFetchContext` for the context.

Reply:
I like both the things you suggested here, so either works for me :)